### ISSUE TITLE
Prior Meds-Separate prior meds from meds prescribed in clinic appointment & revise format for entering prior meds

### ISSUE THREAD

**Issue by MCIR** [christillman/encounterpro_os#65]
When entering a patient’s current medications, client is requesting to be given the option to just enter patients drug(s) name and dosage form 

Their rationale for this approach is, that many patients only remember their medication(s)' name (brand name or ingredient name) and dosage form (oral tablet, nasal spray, ear drops) etc. 
I, on the other hand, foresee a time when they or other users in other clinics may want to document more than just the drug’s name and dosage form. 

I am therefore proposing a two-option approach, where users can choose to select: 
Option 1: Basic entry (search and select the drug’s ingredient and brand name only followed by searching and selecting the drug’s dosage form) or  
Option 2: search and select the drug’s formulation which usually includes the strength and dosage form

In summary when completed
Medications will have 3 different categories. 
1.	‘Past Medications- medications that were entered by physicians as prescriptions on the application, and have expired because they had an expiry date, 
2.	Prior Medications- medication names and dosage forms entered by nurses during triage. I wonder if they will have an expiry date. 
3.	Current Medications which could be renamed ‘Prescribed Medications’ to mean medications that physicians added through the application.

**Comment by MCIR** [comment]
Hi Chris

The prior meds feature works exactly as we discussed. I've successfully incorporated it into the workflows, and the page displays medications as expected. The steps for choosing medications by drug name and dosage form are straightforward.
An interesting behavior I noticed which took me some time to figure out was, in some instances, after selecting a medication name, the application makes the user choose a dosage form before proceeding to duration, while in other instances, the duration selection follows immediately after choosing the medication. This left me questioning why the application wasn’t allowing a dosage form selection. After going through this a couple of times, I realized this behavior was because drugs with only one dosage form automatically default to it, eliminating the need for a separate selection step.

Another observation I made was, it's possible to bypass selecting a dosage form and duration entirely. In these instances, only the medication name shows. While I initially thought this might be an issue, it seems consistent with the legacy application's behavior. I suggest we just monitor this and wait for user feedback to take action.

Lastly, since prior meds and current medications are important sections in a patient summary note, separating them makes it possible to display them as separate sections in the patient summary RTF script. 
To test this in the patient summary RTF script, I added the new "Prior Medications" treatment type and enabled the existing "Prescribed Medications" treatment type. The outcome was, that the existing "Prescribed Medications" section displays medications as expected but the newly added "Prior Medications" treatment type is not displaying prior medications in the patient summary RTF script, despite the patient having prior medications.

A new issue I identified with version 7.2.1.9 is the formatting has changed compared to previous versions I had configured. Is this an unintended result of working on prior meds or part of an ongoing effort to enhance the RTF script format?

The link below is to a video I recorded showing formatting changes I noticed in the patient summary RTF script in version 7.2.1.9. The video also demonstrates the process of adding the "Prior Meds" treatment type and highlights the issue where prior medications aren't displaying despite the command being added.

https://1drv.ms/v/s!AuBDbs81Gi8WgaIjdQbsLzZ-s0Ef_Q?e=Bs6P5o

**Comment by christillman** [comment]
The change in formatting shown in the video was a problem with the final release of 7.2.1.9 because the old RTF control was chosen. This was corrected in 7.2.2.0. I changed the button titles as noted.

As for the Prior Meds section in the script, I will take a look at that. Probably the subscript Standard Treatment Prescription Display is not finding Prior Meds.

**Comment by MCIR** [comment]
Okay thanks

**Comment by christillman** [comment]
Button title changes are in 7.2.2.1
